zephyr-apps/mppt/src/eFuse.c

```diff
@@ -0,0 +1,91 @@
+#include "eFuse.h"
+LOG_MODULE_REGISTER(mppt_eFuse, LOG_LEVEL_INF);
+
+//Defaults pin c14 to active at high voltage
+const struct gpio_dt_spec sig0 = GPIO_DT_SPEC_GET(DT_PATH(zephyr_user), signal_gpios);
+
+//Read E-Fuse pins
+int get_efuse_voltage(void)
+{
+    int ret; 
+    int current_state;
+
+    if (!device_is_ready(sig0.port)) {
+		return -1;
```

Comment:
In general, good practice instead of returning -1 when encountering an error is to return a enum that corresponds to a specific error type: 

https://docs.zephyrproject.org/apidoc/latest/group__system__errno.html